### PR TITLE
lib/worker: add messasge channel between master and worker

### PR DESCRIPTION
**Pull request by amyangfei**
Support pause a running job

- Add a message channel between master and worker, worker registers `WorkerStatusChangeRequestTopic` for status change request from master
- Add ` OnMasterMessage(topic p2p.Topic, message p2p.MessageValue) error` API to `WorkerImpl`
- Add a `nonblocking` parameter in `SendMessage` function of worker handler
- Add `pause-job` command in ctl
- **User logic should handle the status change request message, in job master, it should update job status and call `Exit` when needed. In worker it can stop necessary tasks when the user logic think it is safe to exit.**

### CODE REVIEW

File: pkg/p2p/message_sender.go [hunk 1]
```diff
@@ -28,6 +31,19 @@ func NewMessageSender(router MessageRouter) MessageSender {
...
+// SendToNodeB implements MessageSender.SendToNodeB
+func (m *messageSenderImpl) SendToNodeB(
+	ctx context.Context, targetNodeID NodeID, topic Topic, message interface{},
+) error {
+	client := m.router.GetClient(targetNodeID)
+	if client == nil {
+		return nil
+	}
+
+	_, err := client.SendMessage(ctx, topic, message)
```
Comment (liuzix) [inline pkg/p2p/message_sender.go:43]: The blocking send might have a performance issue. Can you record a TODO?

**Comment by liuzix** [comment]
Even with a blocking send, there is no guarantee that the message is received. For a check for reliable delivery, you need to check the `Ack` status. Can you make sure there is no problem in the use case in this PR?

File: lib/worker.go [hunk 2]
```diff
@@ -406,6 +402,23 @@ func (w *DefaultBaseWorker) initMessageHandlers(ctx context.Context) error {
...
 	}
+
+	topic = WorkerStatusChangeRequestTopic(w.masterID, w.id)
+	ok, err = w.messageHandlerManager.RegisterHandler(
+		ctx,
+		topic,
+		&StatusChangeRequest{},
+		func(sender p2p.NodeID, value p2p.MessageValue) error {
+			msg := value.(*StatusChangeRequest)
+			return w.Impl.OnMasterMessage(topic, msg)
```
Comment (nongfushanquan) [inline lib/worker.go:413]: if the onMasterMessage costs a long time, the message server won't work

**Comment by amyangfei** [comment]
> Even with a blocking send, there is no guarantee that the message is received. For a check for reliable delivery, you need to check the `Ack` status. Can you make sure there is no problem in the use case in this PR?

Is the `ack` status exposed from p2p library, how to check it?

File: lib/worker.go (hunk 2 above)
Comment (amyangfei) [reply lib/worker.go:413]: Good catch, I will add an asynchronous mechanism

**Comment by amyangfei** [comment]
PTAL @nongfushanquan

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**